src/nplinker/pairedomics/podp_antismash_downloader.py

```diff
@@ -25,81 +31,228 @@
+        )
+        # first check if file is cached locally
+        if Path.exists(Path(genome_obj.filename)):
+            # file already downloaded
+            logger.info(
+                f'Genome ID {raw_genome_id} already downloaded to {genome_obj.filename}'
+            )
+            genome_record['resolved_refseq_id'] = genome_obj.resolved_refseq_id
+        elif genome_obj.attempted:
+            # lookup attempted previously but failed
+            logger.info(
+                f'Genome ID {raw_genome_id} skipped due to previous failure')
+            genome_record['resolved_refseq_id'] = genome_obj.resolved_refseq_id
+        else:
+            # if no existing file and no lookup attempted, can start process of
+            # trying to retrieve the data
+
+            # lookup the ID
+            logger.info(
+                f'Beginning lookup process for genome ID {raw_genome_id}')
+
+            assert isinstance(_resolve_refseq_id(genome_record['genome_ID']),
+                              str)
+            genome_obj.resolved_refseq_id = _resolve_refseq_id(
+                genome_record['genome_ID'])
+            genome_obj.attempted = True
+
+            if genome_obj.resolved_refseq_id == "":
+                # give up on this one
+                logger.warning(f'Failed lookup for genome ID {raw_genome_id}')
+                genome_obj.add_line(genome_status_file)
+                continue
+
+        # if we got a refseq ID, now try to download and extract the data from antismash
+        download_and_extract_antismash_data(genome_obj.resolved_refseq_id,
+                                            project_download_cache,
+                                            project_file_cache)
+
+        genome_obj.add_line(genome_status_file)
+
+        output_path = Path(project_file_cache, 'antismash',
+                           genome_obj.resolved_refseq_id)
+        if not Path.exists(Path(output_path, 'completed')):
+            with open(Path(output_path, 'completed'), 'w'):
+                pass
+
+    missing = len([x for x in genome_status.values() if len(x.filename) == 0])
+    logger.info(
+        f'Dataset has {missing} missing sets of antiSMASH data (from a total of {len(genome_records)})'
+    )
+
+    for obj in genome_status.values():
+        obj.add_line(genome_status_file)
+
+    if missing == len(genome_records):
+        logger.warning('Failed to successfully retrieve ANY genome data!')
+
+
+def _get_genome_status_log(
+        genome_status_file: Path) -> dict[str, GenomeStatus]:
+    """Get a dict of GenomeStatus objects by reading given genome status file.
+    Note that a empty dict is returned if the given file does not exist.
+
+    Args:
+        genome_status_file(str | PathLike): it records genome IDs and local filenames to avoid having to repeat HTTP requests
+        each time the app is loaded (this can take a lot of time if there are dozens of genomes).
```

Comment:
```suggestion
        genome_status_file(PathLike): Path to genome status file that records
			genome IDs and local filenames to avoid repeating time-consuming
			HTTP requests each time the app is loaded.
```